spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/auth_key_pair_spec.rb

```diff
@@ -0,0 +1,43 @@
+describe ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::AuthKeyPair do
+  let(:ems) do
+    FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_cloud, :provider_region => "us-south")
+  end
+
+  let(:the_raw_key_pair) do
+    double.tap do |key_pair|
+      allow(key_pair).to receive(:name).and_return('key1')
+      allow(key_pair).to receive(:fingerprint).and_return('0000')
+      allow(key_pair).to receive(:public_key).and_return('AAA...B')
+      allow(key_pair).to receive(:private_key).and_return('BBB...C')
+    end
+  end
+
+  describe 'key pair create and delete' do
+    it 'creates new key pair' do
+      service = double
+      key_pairs = double
+      allow(ExtManagementSystem).to receive(:find).with(ems.id).and_return(ems)
+      allow(ems).to receive(:connect).with(:service => 'PowerIaas').and_return(service)
+      allow(service).to receive(:key_pairs).and_return(key_pairs)
+    end
+
+    it 'deletes existing key pair' do
+      service = double
+      subject.name = 'key1'
+      subject.resource = ems
+      allow(ems).to receive(:connect).with(:service => 'PowerIaas').and_return(service)
+      allow(service).to receive(:delete_key_pair).with('key1')
```

Comment:
Probably should be `expect(server).to receive .....` so that this is testing something

Reply:
Hi Adam while changing this type i am  getting this error
```
Randomized with seed 9594
..FF

Failures:

  1) ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::AuthKeyPair key pair create and delete creates new key pair
     Failure/Error: power_iaas = ext_management_system.connect(:target => "PowerIaas")
     
       #<ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager id: 53000000000085, name: "ems_0000000000002", created_on: "2020-09-21 13:35:35", updated_on: "2020-09-21 13:35:35", guid: "f19b37ef-8a1a-4528-87e9-d526d2057cc7", zone_id: 53000000000029, type: "ManageIQ::Providers::IbmCloud::PowerVirtualServers...", api_version: nil, uid_ems: nil, host_default_vnc_port_start: nil, host_default_vnc_port_end: nil, provider_region: "us-south", last_refresh_error: nil, last_refresh_date: nil, provider_id: nil, realm: nil, tenant_id: nil, project: nil, parent_ems_id: nil, subscription: nil, last_metrics_error: nil, last_metrics_update_date: nil, last_metrics_success_date: nil, tenant_mapping_enabled: nil, enabled: true, options: nil, zone_before_pause_id: nil, last_inventory_date: nil> received :connect with unexpected arguments
         expected: ({:service=>"PowerIaas"})
              got: ({:target=>"PowerIaas"})
       Diff:
       @@ -1 +1 @@
       -[{:service=>"PowerIaas"}]
       +[{:target=>"PowerIaas"}]
       
        Please stub a default value first if message might be received with other args as well. 
     # ./app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/auth_key_pair.rb:5:in `raw_create_key_pair'
     # ./spec/manageiq/app/models/manageiq/providers/cloud_manager/auth_key_pair.rb:43:in `create_key_pair'
     # ./spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/auth_key_pair_spec.rb:22:in `block (3 levels) in <top (required)>'
     # /home/pradeep/.rvm/gems/ruby-2.5.8/gems/webmock-3.9.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

  2) ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::AuthKeyPair key pair create and delete deletes existing key pair
     Failure/Error: expect(server).to receive(:delete_key_pair).with('key1')
     
     NameError:
       undefined local variable or method `server' for #<RSpec::ExampleGroups::ManageIQProvidersIbmCloudPowerVirtualServersCloudManagerAuthKeyPair::KeyPairCreateAndDelete:0x000000000d83c6d0>
     # ./spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/auth_key_pair_spec.rb:30:in `block (3 levels) in <top (required)>'
     # /home/pradeep/.rvm/gems/ruby-2.5.8/gems/webmock-3.9.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'

Finished in 2.65 seconds (files took 8.97 seconds to load)
4 examples, 2 failures

Failed examples:

rspec ./spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/auth_key_pair_spec.rb:16 # ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::AuthKeyPair key pair create and delete creates new key pair
rspec ./spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/auth_key_pair_spec.rb:25 # ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::AuthKeyPair key pair create and delete deletes existing key pair

Randomized with seed 9594
```